src/cluster_legacy.c

```diff
@@ -186,6 +187,54 @@ dictType clusterSdsToListType = {
     NULL                /* allow to expand */
 };
 
+typedef struct {
+    enum { ITER_DICT, ITER_LIST } type;
+    union {
+        dictIterator di;
+        listIter li;
+    };
+} ClusterNodeIterator;
+
+void clusterNodeIterInitAllNodes(ClusterNodeIterator *iter) {
```

Comment:
nit - static? since these are internal functions. we probably don't need explicit inlining though
```suggestion
static void clusterNodeIterInitAllNodes(ClusterNodeIterator *iter) {
```

Reply:
Yeah, we can add static to almost all new functions in this PR.

Reply:
>we can add static to almost all new functions in this PR.

+1. @roshkhatri, can you add static to other new internal functions too?